### PR TITLE
Backwards compatibility & shortcut command changes

### PR DESCRIPTION
**Pull request by joehand**
* `dat <link> {dir}` - stateless clone/sync command, #91 
* `dat {cmd} dir` - dir as third arg for most commands, #90 

TODO: 

- [x] update existing tests for dir args
- [x] add new test for stateless clone command
- [x] decide if mkdirp should happen
- [x] documentation on new command & args

### CODE REVIEW

**Comment by joehand** [comment]
Looks like its just the auth tests failing. Going to fix those in another PR.

This one is set.

**Comment by okdistribute** [comment]
looks good

File: lib/commands/create.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ module.exports = {
 
 function create (opts) {
   opts.resume = false // cannot resume for create
-  if (opts._.length) opts.dir = opts._[0]
+  if (opts._.length && opts.dir === process.cwd()) opts.dir = opts._[0] // use first arg as dir if default set
```
Comment (okdistribute) [inline lib/commands/create.js:34]: does this pattern also work for comands that contain home dirs like `dat sync ~/path/to/mydata`

Comment (joehand) [reply lib/commands/create.js:34]: Yes!

The `&& opts.dir === process.cwd()` part is checking if `--dir` is set (`process.cwd()` is the default). Whatever `opts.dir` gets set to, dat-node resolves it.

**Pull request merged**